plugins/scale/wayfire/plugins/scale-transform.hpp

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include <wayfire/view-transform.hpp>
+#include <wayfire/nonstd/observer_ptr.h>
+#include <wayfire/render-manager.hpp>
+#include <string>
+#include <list>
+#include <algorithm>
+
+namespace wf
+{
+/**
+ * transformer used by scale -- it is an extension of the 2D transformer
+ * with the ability to add overlays
+ */
+class scale_transformer_t : public wf::view_2D
+{
+  public:
+    scale_transformer_t(wayfire_view view) : wf::view_2D(view)
+    {}
+    ~scale_transformer_t()
+    {}
+
+    struct padding_t
+    {
+        unsigned int top    = 0;
+        unsigned int left   = 0;
+        unsigned int bottom = 0;
+        unsigned int right  = 0;
+
+        /**
+         * Compare paddings. Returns true if ANY of the dimensions in this
+         * is smaller than other. Note that this does not define an ordering
+         * on paddings.
+         */
+        bool any_smaller_than(const padding_t& other) const
+        {
+            return (top < other.top) || (left < other.left) ||
+                   (bottom < other.bottom) || (right < other.right);
+        }
+
+        /**
+         * Helper function to extend padding to ensure it is at least as
+         * large as other.
+         */
+        void extend(const padding_t& other)
+        {
+            top    = std::max(top, other.top);
+            left   = std::max(left, other.left);
+            bottom = std::max(bottom, other.bottom);
+            right  = std::max(right, other.right);
+        }
+    };
+
+    uint32_t get_z_order() override
+    {
+        /* ensure that scale comes after fire */
+        return wf::TRANSFORMER_HIGHLEVEL + 2;
```

Comment:
Unfortunately, this can't be. Scale needs to come before wobbly so that dragging a view works, see #1020 

Also, why should scale come after fire? Makes no sense.

Reply:
So, this should be highlevel - 10 or something like that.

Reply:
Ok, that's reasonable. The motivation for putting this after fire was that if an overlay adds padding, then fire will cover the whole (now padded) area -- this can look strange (e.g. if a wide overlay is over a small dialog). But this is a general problem with fire (e.g. fire animation with a 45-degree rotated view will also look strange), so this can be a separate issue.